### PR TITLE
Add table transformer [v2]

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR adds [Table Transformer](https://github.com/microsoft/table-transformer) by Microsoft, as a separate model, rather than tweaking the existing DETR implementation.

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
Thanks a lot for adding this new model!

New model additions are very big PRs on their own, so let's remove the non-related stuff in another PR please.

**Comment by NielsRogge** [comment]
@sgugger I've removed the non-related Table Transformer stuff to #19644

**Comment by NielsRogge** [comment]
@sgugger PR is ready, feel free to approve :)

File: src/transformers/models/table_transformer/modeling_table_transformer.py [hunk 1]
```diff
@@ -0,0 +1,1941 @@
...
+        mask = torch.ones((batch_size, height, width), dtype=torch.bool, device=device)
+        for img, pad_img, m in zip(tensor_list, tensor, mask):
+            pad_img[: img.shape[0], : img.shape[1], : img.shape[2]].copy_(img)
+            m[: img.shape[1], : img.shape[2]] = False
+    else:
+        raise ValueError("Only 3-dimensional tensors are supported")
+    return NestedTensor(tensor, mask)
+
+
+def center_to_corners_format(x):
```
Comment (sgugger) [inline src/transformers/models/table_transformer/modeling_table_transformer.py:1934]: Should probably use the same new variable names as what you did for DETR :-)

**Review by sgugger — APPROVED** [review]
Thanks again for working on this model!

**Pull request merged**